R/zchunk_L221.land_input_1.R

```diff
@@ -54,124 +71,222 @@ module_aglu_L221.land_input_1_DISABLED <- function(command, ...) {
-      add_comments("can be multiple lines") %>%
+
+    # 1. Process inputs
+
+    # Replace GLU names and Add region names
+    L121.CarbonContent_kgm2_R_LT_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) %>%
+      rename(mature.age = `mature age`) ->
+      L121.CarbonContent_kgm2_R_LT_GLU
+
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L125.LC_bm2_R_LT_Yh_GLU
+
+    L131.LV_USD75_m2_R_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) %>%
+      ungroup ->
+      L131.LV_USD75_m2_R_GLU
+
+    # Add region names to inputs
+    L125.LC_bm2_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L125.LC_bm2_R
+
+    A_soil_time_scale_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      A_soil_time_scale_R
+
+    # Convert land value to  1975$ per thousand km2 and calculate minimum land value,
+    # setting a minimum threshold on the land values to ensure that no land use regions get a value of zero
+    L131.LV_USD75_m2_R_GLU %>%
+      mutate(LV_USD75_bm2 = round(LV_USD75_m2 * CONV_BM2_M2, aglu.DIGITS_LAND_VALUE)) ->
+      L131.LV_USD75_bm2_R_GLU
+
+    L131.LV_USD75_bm2_R_GLU %>%
+      select(LV_USD75_bm2) %>%
+      filter(LV_USD75_bm2 > 0) %>%
+      min ->
+      min_LV_USD75_bm2
+
+
+    # 2. Build tables
+
+    # Build L221.LN0_Logit: Logit exponent of the top-level (zero) land nest
+    L125.LC_bm2_R %>%
+      select(region) %>%
+      mutate(LandAllocatorRoot = "root",
+             logit.year.fillout = min(BASE_YEARS),
+             logit.exponent = aglu.N0_LOGIT_EXP) ->
+      L221.LN0_Logit
+
+
+    # Build L221.LN0_Land: Total regional land allocation
+    L125.LC_bm2_R %>%
+      mutate(LandAllocatorRoot = "root",
+            year.fillout = min(BASE_YEARS)) %>%
+      rename(landAllocation = LC_bm2) %>%
+      select(-GCAM_region_ID) ->
+      L221.LN0_Land
+
+
+    # Build L221.LN0_SoilTimeScale: Soil time scale by region
+    A_soil_time_scale_R %>%
+      filter(!(region %in% aglu.NO_AGLU_REGIONS)) %>%
+      select(-GCAM_region_ID) %>%
+      mutate(LandAllocatorRoot = "root") ->
+      L221.LN0_SoilTimeScale
+
+
+    # Build L221.LN1_ValueLogit: Unmanaged land value by region and AEZ, and logit exponent of first nest
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      left_join(select(A_LT_Mapping, Land_Type, LandNode1), by = "Land_Type") %>%
+      select(region, GLU, LandNode1) %>%
+      distinct %>%
+      na.omit %>%
+      mutate(LandAllocatorRoot = "root",
+             logit.year.fillout = min(BASE_YEARS))  %>%
+      # add exponents and logit types
+      left_join(A_LandNode_logit, by = c("LandNode1" = "LandNode")) %>%
```

Comment:
Here and below, I assume there's a reason we're not using `left_join_error_no_match`?